homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,169 @@
+
+    async_add_entities(
+        Alpha2Climate(coordinator, heat_area_id) for heat_area_id in coordinator.data
+    )
+
+
+# https://developers.home-assistant.io/docs/core/entity/climate/
+class Alpha2Climate(CoordinatorEntity, ClimateEntity):
+    """Alpha2 ClimateEntity."""
+
+    coordinator: Alpha2BaseCoordinator
+    target_temperature_step = 0.2
+
+    _attr_supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+    _attr_hvac_modes = [HVAC_MODE_HEAT, HVAC_MODE_COOL]
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_preset_modes = [PRESET_AUTO, PRESET_DAY, PRESET_NIGHT]
+
+    def __init__(self, coordinator: Alpha2BaseCoordinator, heat_area_id: str) -> None:
+        """Initialize Alpha2 ClimateEntity."""
+        super().__init__(coordinator)
+        self.heat_area_id = heat_area_id
+
+    @property
+    def name(self) -> str:
+        """Return the name of the climate device."""
+        return self.coordinator.data[self.heat_area_id]["HEATAREA_NAME"]
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return float(self.coordinator.data[self.heat_area_id].get("T_TARGET_MIN", 0.0))
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return float(self.coordinator.data[self.heat_area_id].get("T_TARGET_MAX", 30.0))
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return float(self.coordinator.data[self.heat_area_id].get("T_ACTUAL", 0.0))
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return current hvac mode."""
+        if self.coordinator.get_cooling():
+            return HVAC_MODE_COOL
+        return HVAC_MODE_HEAT
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        await self.coordinator.async_set_cooling(hvac_mode == HVAC_MODE_COOL)
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation."""
+        if not self.coordinator.data[self.heat_area_id]["_HEATCTRL_STATE"]:
+            return CURRENT_HVAC_IDLE
+        if self.coordinator.get_cooling():
+            return CURRENT_HVAC_COOL
+        return CURRENT_HVAC_HEAT
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature we try to reach."""
+        return float(self.coordinator.data[self.heat_area_id].get("T_TARGET", 0.0))
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperatures."""
+        target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        if target_temperature is None:
+            return
+
+        await self.coordinator.async_set_target_temperature(
+            self.heat_area_id, target_temperature
+        )
+        self.coordinator.data[self.heat_area_id]["T_TARGET"] = target_temperature
```

Comment:
Since we're calling the callbacks before we modify the coordinator data here, the entities won't see this update. Maybe move this update to the coordinator data inside the coordinator?

Reply:
You are right, much better.